dask/dataframe/optimize.py

```diff
@@ -65,28 +68,47 @@ def optimize_read_parquet_getitem(dsk):
                 block_columns = [block_columns]
 
             columns |= set(block_columns)
+            update_blocks[dep] = block
 
         old = layers[k]
 
         if columns and columns < set(old.meta.columns):
             columns = list(columns)
             meta = old.meta[columns]
+            name = "read-parquet-" + tokenize(old.name, columns)
```

Comment:
But what about dependent layers?  They might still refer to the old name, no?

Reply:
Ah, I see, this is what `update_blocks` is about.  What if one of the dependents isn't a Blockwise?

Reply:
That's my primary concern right now... It seems like these changes should break that, right?

My inclusion of the `df2c`, which used `.rename` was an intent to cover this case but I'm not sure if it did.

Reply:
Why can't we keep the old name?

We might also not run this optimization if there are any non-Blockwise dependents.

As a test, you might try calling something like a rolling aggregation or repartition afterwards, which wouldn't be a blockwise.

Reply:
From what I can tell, the optimization is disabled in that case.


```python
In [18]: r = df2[["a", "b"]].rolling(3).max() + df2.rolling(3).max()

In [19]: dict(r.dask)
Out[19]:
{('add-10edb0867bce48f1be2892de12e08cda', 0): (subgraph_callable,
  ('max-8d08edfeb5d3400a5bdd623b5deaaad6', 0),
  ('max-35c8931b57ab0ea95efa92dea4d5893c', 0)),
 ('max-8d08edfeb5d3400a5bdd623b5deaaad6', 0): (subgraph_callable,
  ('getitem-f507c0044e30b36d995ca7fb19f3a5c0', 0),
  {'window': 3,
   'min_periods': None,
   'center': False,
   'win_type': None,
   'axis': 0},
  'max',
  'getitem-f507c0044e30b36d995ca7fb19f3a5c0'),
 ('getitem-f507c0044e30b36d995ca7fb19f3a5c0', 0): (subgraph_callable,
  ('read-parquet-03418aac4452358f237f92b12cbed13d', 0),
  ['a', 'b']),
 ('read-parquet-03418aac4452358f237f92b12cbed13d',
  0): (<function dask.dataframe.io.parquet.core.read_parquet_part(func, fs, meta, part, columns, index, kwargs)>, <function dask.dataframe.io.parquet.arrow.ArrowEngine.read_partition(fs, piece, columns, index, categories=(), partitions=(), **kwargs)>, <fsspec.implementations.local.LocalFileSystem at 0x11904e940>, Empty DataFrame
  Columns: [a, b, c, d]
  Index: [], [('/tmp/path.parquet', None, [])], ['a',  # <------------------- Notice this isn't `[a, b]`.
   'b',
   'c',
   'd'], None, {'partitions': None, 'categories': None}),
 ('max-35c8931b57ab0ea95efa92dea4d5893c', 0): (subgraph_callable,
  ('read-parquet-03418aac4452358f237f92b12cbed13d', 0),
  {'window': 3,
   'min_periods': None,
   'center': False,
   'win_type': None,
   'axis': 0},
  'max',
  'read-parquet-03418aac4452358f237f92b12cbed13d')}
```

I believe that's because all the early `return`s in `optimize_read_parquet_getitem`. As soon as we see an operation other than `__getitem__`, we bail out on the optization.